samples/algorithms/BitFlip.qs

```diff
@@ -0,0 +1,139 @@
+
+        // Set the initial state of the physical qubit.
+        SetSampleState(physicalQubit);
+
+        // Using two additional qubits, encode the physical qubit into a
+        // logical qubit.
+        use aux = Qubit[2];
+        EncodeAsLogicalQubit(physicalQubit, aux);
+        let logicalQubit = [physicalQubit] + aux;
+
+        // Induce a bit-flip error on a random qubit.
+        X(logicalQubit[DrawRandomInt(0, 2)]);
+
+        // Show the logical qubit with the error state.
+        DumpMachine();
+
+        // Find and correct the bit-flip error.
+        CorrectError(logicalQubit);
+
+        // Show the logical qubit with the corrected state.
+        DumpMachine();
+
+        // Decode the logical qubit back into a single physical qubit.
+        Adjoint EncodeAsLogicalQubit(physicalQubit, aux);
+
+        // Measure and reset the physical qubit before releasing it.
+        let result = M(physicalQubit);
+        Reset(physicalQubit);
+        return result;
+    }
+
+    /// # Summary
+    /// This operation sets the state of the given qubit such that
+    /// it will have a 20% likelihood of resulting in a `Zero` and
+    /// 80% likelihood of resulting in a `One` when measured in the
+    /// computational basis. The input qubit is expected to be in
+    /// the |0〉 state.
+    ///
+    /// # Input
+    /// ## q
+    /// The given qubit to be put into superposition. It is assumed that this
+    /// qubit is in its default |0〉 state.
+    operation SetSampleState(q : Qubit) : Unit {
+        let alpha = 0.20;
+        Ry(2.0 * ArcCos(Sqrt(alpha)), q);
+    }
+
+    /// # Summary
+    /// This operation takes the given `physicalQubit` state,
+    /// (α|0〉 + β|1〉) / √2, and encodes it in the `aux` qubits. This
+    /// encodes all the qubits into a single logical qubit whose state reflects
+    /// the state of the given `physicalQubit`: (α|000〉 + β|111〉) / √2.
+    ///
+    /// # Input
+    /// ## physicalQubit
+    /// The qubit whose state, (α|0〉 + β|1〉) / √2, is to be encoded in the
+    /// logical qubit.
+    ///
+    /// ## aux
+    /// The auxiliary qubits that will be used as part of the encoding. These
+    /// should be grouped with the `physicalQubit` to form the logical qubit.
+    operation EncodeAsLogicalQubit(physicalQubit : Qubit, aux : Qubit[]) : Unit is Adj {
+        ApplyToEachA(CNOT(physicalQubit, _), aux);
+    }
+
+    /// # Summary
+    /// This operation detects and corrects a single bit-flip error for a logical
+    /// qubit encoded as three physical qubits. When finished, the given register
+    /// of qubits will be in the state: (α|000〉 + β|111〉) / √2.
+    ///
+    /// # Input
+    /// ## physicalQubits
+    /// The given register of three physical qubits representing a single logical qubit
+    /// having superposition (α|0〉 + β|1〉) / √2.
+    /// This logical qubit can have up to one bit-flip error that will be corrected.
+    operation CorrectError(physicalQubits : Qubit[]) : Unit {
```

Comment:
`physicalQubits` as a parameter name is a bit confusing since we refer to this as a logical qubit in the `Main` operation. Changing the name to `logicalQubit` would make it more clear IMO.

Reply:
done